### ISSUE TITLE
testSSD300.py  have no error but stop

### ISSUE THREAD

**Issue by lfydegithub** [Stick-To/SSD-tensorflow#2]
when i run testSSD300.py, it broken with anything, no error imformation， just stop Quietly ， ：）....

> E:\pythonWork\ST-ssd
(py36) λ python testSSD300.py
['/data/train_00001-of-00010.tfrecord', '/data/train_00002-of-00010.tfrecord', '/data/train_00003-of-00010.tfrecord', '/data/train_00004-of-00010.tfrecord', '/data/train_00005-of-00010.tfrecord', '/data/train_00006-of-00010.tfrecord', '/data/train_00007-of-00010.tfrecord', '/data/train_00008-of-00010.tfrecord', '/data/train_00009-of-00010.tfrecord', '/data/train_00010-of-00010.tfrecord']
WARNING:tensorflow:From D:\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\ops\sparse_ops.py:1165: sparse_to_dense (from tensorflow.python.ops.sparse_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Create a `tf.sparse.SparseTensor` and use `tf.sparse.to_dense` instead.
D:\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\ops\gradients_impl.py:112: UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory.
  "Converting sparse IndexedSlices to a dense Tensor of unknown shape. "
------------------------- epoch 0 -------------------------
1
2140
2
E:\pythonWork\ST-ssd
(py36) λ

with train_one_epoch（）is :

> def train_one_epoch(self, lr):
        self.is_training = True
        self.sess.run(self.train_initializer)
        print('1')
        mean_loss = []
        num_iters = self.num_train // self.batch_size
        print(num_iters)
        for i in range(num_iters):
            print('2')
            _, loss = self.sess.run([self.train_op, self.loss], feed_dict={self.lr: lr})
            print('3')
            sys.stdout.write('\r>> ' + 'iters '+str(i)+str('/')+str(num_iters)+' loss '+str(loss))
            sys.stdout.flush()
            mean_loss.append(loss)
        sys.stdout.write('\n')
        mean_loss = np.mean(mean_loss)
        return mean_loss

  
why？  can u help me？ very very very thx

**Comment by lfydegithub** [comment]
I know the reason，
[because tf.gether() 's input has ‘None’ in shape](https://stackoverflow.com/questions/35892412/tensorflow-dense-gradient-explanation)
above website suggest that tf.dynamic_partition()  instead of tf.gather(),, but how to use...

**Comment by Stick-To** [comment]
This output is a warning tip, not an error. It would not influence the true result. If you need help, give me the error tips. Or you could try this code on voc dataset, There is no problem occuring on my computer.
